library/public/utils.h

```diff
@@ -57,6 +61,94 @@ class F3D_EXPORT utils
   {
     explicit tokenize_exception(const std::string& what = "");
   };
+
+  /** String template allowing substitution of variables enclosed in curly braces.
+    ```
+    StringTemplate("{greeting} {name}!")
+      .substitute({ { "greeting", "hello" }, { "name", "World" } })
+      .str() == "hello World!"
+    ```
+   */
+  class StringTemplate
```

Comment:
All our public class and types (not enums) are snake case. This should be named `string_template` I believe